crepecake/src/main/java/com/google/cloud/tools/crepecake/hash/ByteHashBuilder.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.hash;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * Generates SHA-256 hashes in hexadecimal. Can act as an {@link OutputStream} that captures the
+ * bytes to hash.
+ */
+public class ByteHashBuilder extends OutputStream {
```

Comment:
You may be reinventing java.security.DigestOutputStream which implements SHA-256